harmonica/gravity_corrections.py

```diff
@@ -69,3 +74,72 @@ def bouguer_correction(topography, density_crust=2670, density_water=1040):
     density[oceans] = -1 * (density_water - density_crust)
     bouguer = 1e5 * 2 * np.pi * GRAVITATIONAL_CONST * density * topography
     return bouguer
+
+def atmospheric_correction(topography):
+    """
+    Calculates atmospheric correction as per Hinze 2005 eqn 5.
+    
+    .. math::
+
+        g_{atm} = 0.874 - 9.9E-5 h + 3.56E-9 h^2 
+
+    in which :math:`h` is the elevation of the station above the ellipsoid
+    and :math:`g_{atm}` is the gravitational effect of the atmospheric the air
+    mass in mGal.
+
+    Parameters
+    ----------
+    topography : array or :class:`xarray.DataArray`
+        Topography height and bathymetry depth in meters.
+        Should be referenced to the ellipsoid (ie, geometric heights).
+        
+
+    Returns
+    -------
+    atmos_correction : array or :class:`xarray.DataArray`
+    The gravitational effect of the atmosphere in mGal
+
+    """
+    atmos_correction = 0.874 - 9.9e-05 * topography + 3.56e-09 * topography**2
+    
+    return atmos_correction
+
+
+def spherical_bouguer_cap_correction(topography):
+    """
+    Calculates spherical cap correction for the Bouguer slab as per Hinze 2013.
+    
+    Gravity and Magnetic Exploration, Cambridge Press 2013 
+    
+    .. math::
+
+        g_{spher} =  0.001464139 h - 3.533047e-07 h^{2} + 1.002709e-13 h^{3}
+        + 3.002407E-18 h^{4} 
+
+    in which :math:`h` is the elevation of the station above the ellipsoid
```

Comment:
This strikes me as a bit odd since the effect of a cap depends on 2 things: the height of the observation point and the thickness of the cap (i.e., height of topography). Is this equation assuming that they are the same?

In fact, it should actually depend on the radial coordinate for both of these, not really the heights (which the authors probably assumed as some mean Earth radius). This will cause the equation to fail for lunar or martian applications, where the reference radius is much smaller.

This can quickly get very complicate if we assume that the heights are relative to the ellipsoid not a sphere. One compromise could be:

* We assume a spherical approximation for this corrections. Height should be reference to the ellipsoid but we'll approximate it for a sphere with a given radius. The differences should be relatively small.
* Require passing the radius of this sphere. For the Earth, we can pass `boule.WGS84.mean_radius` easily. It might be best to not default to this value realatively sas it would introduce errors if users forget to specify it when applying to other planets.
 * Allow passing the extent of the cap, which could default to 167 km.

Reply:
The spherical bouguer formula from LaFehr deals with this I think.  In that formulation there is an alpha constant which is the cap radius (i.e. 167 km) divided by the mean radius of Earth.  So both those parameters are tunable for Earth or other planets i'd think. You could add different planet radii to boule and then just call from Harmonica as you suggest.

Reply:
See comments above about using the full analytic expression. I think if we use that then we can include both the planet radius and cap extent (though there is pretty good evidence that beyond 167 km it gets more imprecise as average crustal densities are likely to vary over large scales.) See Figure 4 in LaFehr 1991.  

The full expression does assume a constant radius and is most valid at mid latitudes where radius changes are small.  If we have a method of calculating the specific raidus at the station location, then that could be used instead, but Figure 6 from LaFehr shows that it's only at high altitudes (4000m +) that the effect becomes more than instrument error (i.e max of ~20 uGal difference between pole and equator at 4000m), so unless you have a survey that covers the hemisphere I don't think it matter.

Reply:
I'm still a bit confused here about the radius referred in the papers. Is that assuming that observations are located at the surface of topography? In that case, the radius of the observation point and the radius for the spherical topography would be the same. But if we have observations from airplanes or satellites, this wouldn't work. Right?

But then again, the Bouguer correction itself doesn't work in these cases anyway.

Either way, I like the full analytic expression! It's good to have these things be explicit. I just had a bug in some code doing a Bouguer correction for Mars and forgetting to set `density_water=0` since there are no oceans on Mars. Having defaults can be very dangerous.

Reply:
I'm not too familiar with airborne data processing (and even less satellite) but i did some googling and it seems airborne gravity is processed similar to ground gravity (except it requires the EotVos correction of course, which we could include as a separate function). See report below for processing steps:

http://www.geosciencebc.com/i/project_data/QUESTdata/GBCReport2008-8/Gravity_Technical_Report.pdf

There is a "curvature" correction that requires the input of the **ground surface** (not the aircraft altitude).  When I computed this it is approximately the same as the spherical_bouguer_cap_correction, (0.64368 mGal). 

So i think that for airborne data you would calculate the "normal gravity" at the aircraft elevation (which is where the sensor is) and then calculate the spherical bouguer slab at the elevation of the ground below the aircraft (which can be dervied from the laser altimeter data etc (flight height - laser altimeter = ground height). 

In summary, it seems we can use the full spherical bouguer for both ground and airborne gravity. The user needs to ensure they use the correct elevations (aircraft vs ground) for the various parts of the complete Bouguer Anomaly formulation.

Reply:
> In summary, it seems we can use the full spherical bouguer for both ground and airborne gravity. The user needs to ensure they use the correct elevations (aircraft vs ground) for the various parts of the complete Bouguer Anomaly formulation.

Yes. The problem I mentioned is that gravity should decay with observation height but in all of these different Bouguer corrections the observation height isn't a parameter. That is because the Bouguer plate has constant gravity with height due to being infinite. This is an OK approximation if the observation is on top of the topography, but it's an over-correction when the observations are at greater height (airborne).

I was wondering if the spherical version took this into account. But apparently it doesn't.

> The user needs to ensure they use the correct elevations (aircraft vs ground) for the various parts of the complete Bouguer Anomaly formulation.

Yes! This is why we're avoiding the word "height" when talking about Bouguer and using "topography" instead to make it clear that this is not the observation height but the topographic height.